ecosystem/typescript/sdk/examples/typescript/marketplace.ts

```diff
@@ -0,0 +1,403 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable max-len */
+
+import {
+  AptosClient,
+  AptosAccount,
+  HexString,
+  MaybeHexString,
+  OptionalTransactionArgs,
+  Provider,
+  TransactionBuilderRemoteABI,
+} from "aptos";
+
+type TransactionPayload = {
+  type: string;
+  function: string;
+  type_arguments: string[];
+  arguments: any[];
+};
+
+const APTOS_COIN: string = "0x1::aptos_coin::AptosCoin";
+const COIN_LISTING: string = "coin_listing";
+const COLLECTION_OFFER: string = "collection_offer";
+const FEE_SCHEDULE: string = "fee_schedule";
+const LISTING: string = "listing";
+
+/**
+ * Class for managing the example marketplace.  It builds payloads to be used with the wallet adapter, but can
+ * also submit payloads directly with an AptosAccount.
+ */
+export class Marketplace {
+  readonly provider: Provider;
+  readonly code_location: HexString;
+
+  constructor(provider: Provider, code_location: MaybeHexString) {
+    this.provider = provider;
+    this.code_location = HexString.ensure(code_location);
+  }
+
+  // Coin listing operations
+  async initFixedPriceListing(
```

Comment:
all functions dont need to by async and return Promise - just regular functions that return `TransactionPayload`
```suggestion
  initFixedPriceListing(
```